src/controllers/userController.js

```diff
@@ -7,66 +7,63 @@ const {
 } = require('../errors');
 
 const getUserDetails = async (req, res, next) => {
-  const user = await User.findById(req.params.id);
-  if (!user) {
-    throw new NotFoundError(`No user with id ${req.params.id}`);
-  } else {
-    res.status(StatusCodes.OK).json({ user });
-  }
+  res.status(StatusCodes.OK).json({ user: req.user });
```

Comment:
Your req.user object only contains the following properties: name, userId and role which you send back in the response. I suspect the front end would like to see more information about the user, like an email or even favorite pets list. I would discuss this with your front end team. The version of the similar route you had before included more information which frontend team was using on their end.